ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,295 @@
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.deregister_instances(instance_list)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Instance not removed from Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} removed from Load Balancer {1}.'
+        .format(instance_id, elb_name))
+    _remove_instance_from_elb_list_in_properties(instance_id)
+
+
+@operation
+def add_instance_to_elb(**_):
+
+    ctx.logger.info('Attempting to add_instance_to_elb.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.register_instances(instance_id)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+            raise NonRecoverableError('Instance not added to Load Balancer. '
+                                      '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} added to Load Balancer {1}.'
+        .format(instance_id, elb_name))
+
+    _add_instance_to_elb_list_in_properties(instance_id)
+
+
+def _add_health_check_to_elb(elb, health_check):
+
+    hc = _create_health_check(health_check)
+
+    try:
+        elb.configure_health_check(hc)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Health check not added to Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Health check added to Load Balancer {0}.'
+        .format(elb.name))
+
+
+def use_external_elb(**_):
+
+    utils.use_external_resource(ctx.node.properties)
+
+
+@operation
+def delete_elb(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        utils.unassign_runtime_property_from_resource(
+            constants.EXTERNAL_RESOURCE_ID, ctx.instance)
+        return
+
+    ctx.logger.info('Attempting to delete Load Balancer.')
+
+    elb_name = ctx.node.properties['elb_name']
+
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.delete()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer {0} not deleted. '
+                                  .format(str(e)))
+
+    ctx.logger.info(
+        'Load Balancer {0} deleted. '
+        .format(elb_name))
+    if 'elb_name' in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return
```

Comment:
If I remember correctly @EarthmanT said this was the current expected behaviour so I played along. before that it would return True if it actually deleted something and false if it didn't.